source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,535 @@
+in the following 3 categories:
+
+- Connections checked out for cursors
+- Connections checked out for transactions
+- Connections checked out for operations not falling under the previous 2 categories
+
+When the connection pool's :code:`maxPoolSize` is reached and the pool times out waiting
+for a new connection the :code:`WaitQueueTimeoutError` MUST include a new detailed message,
+"Timeout waiting for connection from the connection pool. maxPoolSize: n, connections
+in use by cursors: n, connections in use by transactions: n, connections in use by other
+operations: n". A :code:`ConnectionCheckOutFailedEvent` MUST also be published with the
+same message.
+
+
+Error Handling
+--------------
+
+Initial Handshake Errors
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When establishing a new connection in load balanced mode, drivers MUST NOT perform SDAM
+error handling for any errors that occur before the MongoDB Handshake
+(i.e. :code:`hello` or legacy hello command) is complete. Errors during the MongoDB
+Handshake MUST also be ignored for SDAM error handling purposes. Once the initial
+handshake is complete, the connection MUST determine its generation number based
+on the :code:`serviceId` field in the handshake response. Any errors that occur
+during the rest of connection establishment (e.g. errors during authentication commands)
+MUST go through the SDAM error handling flow but MUST NOT mark the server as
+:code:`Unknown`.
+
+Post-Handshake Errors
+^^^^^^^^^^^^^^^^^^^^^^
+
+When the driver is operating in load balanced mode and an application operation receives a
+state change error, the driver MUST NOT make any changes to the :code:`TopologyDescription`
+or the :code:`ServerDescription` of the load balancer (i.e. it MUST NOT mark the load
+balancer as :code:`Unknown`). If the error requires the connection pool to be cleared,
+the driver MUST only clear connections with the same :code:`serviceId` as the connection
+which errored.
+
+
+Events
+------
+
+When in load balancer mode the driver MUST now include the :code:`serviceId` in the
+:code:`CommandStartedEvent`, :code:`CommandSucceededEvent`, and
+:code:`CommandFailedEvent`. The driver MAY decide how to expose this information.
+Drivers that have a :code:`ConnectionId` object for example, MAY choose to provide a
+:code:`serviceId` in that object. The :code:`serviceId` field is only present when
+in load balancer mode and connected to a service that is behind a load balancer.
+
+Additionally the :code:`PoolClearedEvent` MUST also contain a :code:`serviceId`
+field.
+
+
+Downstream Visible Behavioral Changes
+-------------------------------------
+
+Services MAY add a command line option or other configuration parameter, that tells the service
+it is running behind a load balancer. Services MAY also dynamically determine whether they are
+behind a load balancer.
+
+When the server supports the :code:`loadBalanced` option, the server MUST respond to the command
```

Comment:
This information is duplicated in the fourth paragraph. I recommend removing this paragraph because the other is more descriptive.

Reply:
Removed.